homeassistant/components/tplink/config_flow.py

```diff
@@ -40,27 +69,114 @@ async def async_step_integration_discovery(
     ) -> FlowResult:
         """Handle integration discovery."""
         return await self._async_handle_discovery(
-            discovery_info[CONF_HOST], discovery_info[CONF_MAC]
+            discovery_info[CONF_HOST],
+            discovery_info[CONF_MAC],
+            discovery_info[CONF_DEVICE_CONFIG],
+        )
+
+    @callback
+    def _update_config_if_entry_in_setup_error(
+        self, entry: ConfigEntry, host: str, config: dict
+    ) -> None:
+        """If discovery encounters a device that is in SETUP_ERROR update the device config."""
+        if entry.state != ConfigEntryState.SETUP_ERROR:
+            return
+        entry_data = entry.data
+        entry_config_dict = entry_data.get(CONF_DEVICE_CONFIG)
+        if entry_config_dict == config and entry_data[CONF_HOST] == host:
+            return  # pragma: no cover
```

Comment:
We need to figure out how to cover this line or refactor if it's truly unreachable. HA standards require 100% coverage for config flows.